contracts/AssetPool.sol

```diff
@@ -12,11 +12,11 @@
 
 // SPDX-License-Identifier: MIT
 
-pragma solidity <0.9.0;
+pragma solidity 0.8.4;
 
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
-import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
```

Comment:
Do we still need it?

Reply:
Nope, [Solidity docs](https://docs.soliditylang.org/en/latest/080-breaking-changes.html#how-to-update-your-code) tells to replace it. Done in c154802
However, one file is still untouched `ERC20WithPermit.sol` which includes custom error messages for `.sub()`. Do we want to refactor that as well along with the tests?

Reply:
https://github.com/keep-network/coverage-pools/pull/107/commits/eb8d0546c3bf58d1b065baee638d7f40d830b414 😜 

We do want to keep the same revert messages. I started looking into it, then it escalated... and I figured I'll just commit and leave the review of this change to you.